contentcuration/contentcuration/assets/shared/views/ThumbnailUpload.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <div
+    :style="highlight && !readonly ? { borderColor: 'purple' } : {}"
+    @dragenter.prevent="enter"
+    @dragover.prevent="over"
+    @dragleave.prevent="leave"
+    @drop.prevent="drop"
+  >
+    <template
+      v-for="file in files"
+    >
+      <VProgressCircular
+        v-if="file.progress < 100"
+        :key="file.id"
+        :size="50"
+        :width="15"
+        :value="file.progress"
+      >
+        {{ file.progress }}
+      </VProgressCircular>
+    </template>
+    <input
+      v-if="!readonly"
+      id="fileElem"
+      ref="fileUpload"
+      style="display: none;"
+      type="file"
+      accept="image/*"
+      @change="handleFiles($event.target.files)"
+    >
+    <label for="fileElem">
+      <VImg v-if="file" contain :src="file" />
+      <VImg v-else contain :src="require('../images/kolibri-logo.svg')" />
+      <VBtn v-if="!readonly" @click="$refs.fileUpload.click()">
+        {{ $tr('fileUpload') }}
+      </VBtn>
+    </label>
+  </div>
+</template>
+
+<script>
+
+  import client from 'shared/client';
+
+  export default {
+    name: 'ThumbnailUpload',
+    $trs: {
+      fileUpload: 'Upload file',
+    },
+    props: {
+      value: {
+        type: Object,
+        default() {
+          return {};
+        },
+      },
+      readonly: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    data() {
+      return {
+        highlight: false,
+        files: [],
+      };
+    },
+    computed: {
+      file() {
+        if (this.files.length && this.files[0].previewSrc) {
+          return this.files[0].previewSrc;
+        }
+        return this.value.thumbnail_url;
+      },
+    },
+    methods: {
+      enter() {
+        this.highlight = true;
+      },
+      over() {
+        this.highlight = true;
+      },
+      leave() {
+        this.highlight = false;
+      },
+      drop(e) {
+        this.highlight = false;
+        this.handleFiles(e.dataTransfer.files);
+      },
+      handleFiles(files) {
+        if (!readonly) {
+          [...files].forEach(file => {
+            const id = String(Math.random()).slice(2);
```

Comment:
Generate a random id for the file upload so that we can manipulate the list of uploads later, and have a key for v-for.